llama_index/embeddings/mistralai.py

```diff
@@ -0,0 +1,82 @@
+"""MistralAI embeddings file."""
+
+from typing import Any, List, Optional
+
+from llama_index.bridge.pydantic import Field
+from llama_index.callbacks.base import CallbackManager
+from llama_index.embeddings.base import DEFAULT_EMBED_BATCH_SIZE, BaseEmbedding
+
+
+class MistralAIEmbedding(BaseEmbedding):
+    """Class for MistralAI embeddings.
+
+    Args:
+        model_name (str): Model for embedding.
+            Defaults to "mistral-embed".
+
+        api_key (Optional[str]): API key to access the model. Defaults to None.
+    """
+
+    # Instance variables initialized via Pydantic's mechanism
+    mistralai_client: Any = Field(description="MistralAI client")
```

Comment:
Can we make this `_mistralai_client: Any = PrivateAttr()`

and then in the init

```
        self._mistralai_client = MistralClient(api_key=api_key),
        super().__init__(
            model_name=model_name,
            embed_batch_size=embed_batch_size,
            callback_manager=callback_manager,
            **kwargs,
        )
```

Reply:
Yes. Sure.